modules/dnn/test/test_onnx_importer.cpp

```diff
@@ -2665,17 +2690,84 @@ static void testYOLO(const std::string& weightPath, const std::vector<int>& refC
     std::vector<int> keep_idx;
     NMSBoxes(boxes, confidences, conf_threshold, iou_threshold, keep_idx);
 
-    std::vector<int> keep_classIds;
-    std::vector<float> keep_confidences;
-    std::vector<Rect2d> keep_boxes;
+    FilteredDetections result;
     for (auto i : keep_idx)
     {
-        keep_classIds.push_back(classIds[i]);
-        keep_confidences.push_back(confidences[i]);
-        keep_boxes.push_back(boxes[i]);
+        result.keep_classIds.push_back(classIds[i]);
+        result.keep_confidences.push_back(confidences[i]);
+        result.keep_boxes.push_back(boxes[i]);
     }
+    return result;
+}
+
+
+TEST_P(Test_ONNX_nets, YOLOx)
+{
+    std::string weightPath = _tf("models/yolox_s_inf_decoder.onnx", false);
+
+    Size targetSize{640, 640};
+    float conf_threshold = 0.50;
+    float iou_threshold = 0.50;
 
-    normAssertDetections(refClassIds, refScores, refBoxes, keep_classIds, keep_confidences, keep_boxes);
+    std::vector<int> refClassIds{1, 16, 7};
+    std::vector<float> refScores{0.96f, 0.91f, 0.66f};
+
+    // [x1, y1, x2, y2]
+    std::vector<Rect2d> refBoxes{
+        Rect2d(104.62, 181.28, 470.95, 428.22),
+        Rect2d(112.32, 264.88, 258.11, 527.31),
+        Rect2d(389.63, 144.63, 576.66, 223.18),
+        };
+
+    Image2BlobParams imgParams(
+        Scalar::all(1),
+        targetSize,
+        Scalar::all(0),
+        true,
+        CV_32F,
+        DNN_LAYOUT_NCHW,
+        DNN_PMODE_LETTERBOX,
+        Scalar::all(144)
+        );
+
+    testYOLO(
+        weightPath, refClassIds, refScores, refBoxes,
+        imgParams, conf_threshold, iou_threshold,
+        1.0e-1, 1.0e-1);
```

Comment:
Is that an observed error or just to cover all backends and HW targets? Just seems too big.

Reply:
I only checked on 1 image for all backends. Did not try differences HWs. Why too big? 0.1 is not even a pixel difference.

Reply:
Not agree. 0.1 score threshold means 0.4 and 0.5 prediction score are the same. 0.1 IoU threshold means boxes with 90% overlap or higher are accepted (which is OK, but this is not a single pixel difference).

Reply:
Can you please share confidence scores as well?

Reply:
Looks like NMS works differently in GT and our implementation. What if we drop NMS and just sort boxes by confidence on both sides?

Reply:
> Can you please share confidence scores as well?

conf, xmin, ymin, xmax, ymax

**Target:** 

```
0.96, 104.62, 181.28, 470.95, 428.22
0.91, 112.32, 264.88, 258.11, 527.31
0.66, 389.63, 144.63, 576.66, 223.18
```

**Source:** 

```
0.964937 105.568 179.377 470.586 428.593
0.916445 111.434 263.394 258.764 525.903
0.694355 389.142 143.935 578.056 221.994
```

Reply:
> Looks like NMS works differently in GT and our implementation. What if we drop NMS and just sort boxes by confidence on both sides?

How close must detectors final output match the GT ?

Reply:
> Looks like NMS works differently in GT and our implementation. What if we drop NMS and just sort boxes by confidence on both sides?

@asmorkalov, `normAssertDetections` is flexible - it can match detections in any order. The problem of the thread is too big thresholds. GT is fine.

Reply:
@dkurt if I make the threshold tighter (1.0e - 2) test fails.